packages/financial-templates-lib/src/price-feed/DefiPulseTotalTVLPriceFeed.js

```diff
@@ -0,0 +1,164 @@
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(
+    logger,
+    web3,
+    apiKey,
+    lookback,
+    networker,
+    getTime,
+    minTimeBetweenUpdates,
+    decimals = 18,
+
+  ) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.apiKey = apiKey;
+
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.toBN = this.web3.utils.toBN;
+
+    this.historicalPrices = [{timestamp:0, value:0}];
+
+    this.convertDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), decimals).toString());
+    };
+
+    this.removeValuesOlderThanLookback = () => {
+
+        const currentTime = this.getTime();
+
+        this.historicalPrices = this.historicalPrices.filter(priceObj => priceObj.timestamp > (currentTime - lookback) );
+
+    }
+
+
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    let closestTime = {timestamp:0, value:0};
+
+    //go through all values and find time that that is the largest and still less than 'time'
+    for(let i = 0; i < this.historicalPrices.length; i++){
+
+      let past = this.historicalPrices[i].timestamp;
+      let val = this.historicalPrices[i].value;
+
+      if( (past > closestTime.timestamp) && (past < time)){
+
+        closestTime.timestamp = past;
+        closestTime.value = val;
+
+      }
+
+    }
+
+    const returnPrice = closestTime.value;
+
+
+    return returnPrice;
+  }
+
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "DEFI_PULSE_TOTAL_TVL_PriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "DEFI_PULSE_TOTAL_TVL_PriceFeed",
+      message: "Updating",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // 1. Construct URLs.
+    const priceUrl =
+    `https://data-api.defipulse.com/api/v1/defipulse/api/MarketData?api-key=${this.apiKey}`;
+
+    // 2. Send requests.
+    const [priceResponse] = await Promise.all([
```

Comment:
nit: no point in using `Promise.all` synthax here seeing you are only sending out one async call.